hook.php

```diff
@@ -79,3 +79,31 @@ function plugin_credit_getDropdown() {
                                     Session::getPluralNumber(),
                                     'credit')];
 }
+
+function plugin_credit_get_datas(NotificationTargetTicket $target) {
+   global $DB;
+   $target->data['##lang.credit.voucher##']=__('Credit voucher', 'credit');
+   $target->data['##lang.credit.used##']=__('Quantity consumed', 'credit');
+   $target->data['##lang.credit.left##']=__('Quantity remaining', 'credit');
+   $id=$target->data['##ticket.id##'];
+   $query = "SELECT gpce.id as id,
```

Comment:
This request is really complex.
I think it should work using only following request
```sql
SELECT
   `glpi_plugin_credit_entities`.`name`,
   `glpi_plugin_credit_entities`.`quantity`,
   SUM(`glpi_plugin_credit_tickets.consumed`)
FROM
   `glpi_plugin_credit_entities`,
INNER JOIN
   `glpi_plugin_credit_tickets` ON `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = `glpi_plugin_credit_entities`.`id`
WHERE
   `glpi_plugin_credit_tickets`.`tickets_id` = {$id}
```
and then compute the rest using a simple operation like `'##credit.left##' => $credit['quantity'] - $credit['consumed']`.

Reply:
My intention is that 'credit.used' shows only the credits used on the specific ticket.
I think your example returns  the amount consumed in all the tickets

Reply:
I just checked (after having fixing typos and added GROUP BY clause) and it does not count consumed from other tickets.

Reply:
Hi @cedric-anne ,
This is exactly the idea, although the client should know how many credits have consumed that particular ticket, they should also know how many credits are left. So:

##credit.used## (Quantity consumed in that particular ticket)
##credit.left## (Total quantity remaining )

With your query, we can get ##credit.used## (credits consumed in this ticket), **but we cannot know ##credit.left## (total credits available from this voucher)**.

Example:
Voucher - A - 10 credits
Ticket 1 - Consumed 2
Ticket 2 - Consumed 3

Ticket 2 notification, with your query, would send 7
##credit.used## = 3
##credit.left##' => $credit['quantity'] - $credit['consumed']
              7           =>              10              -           3

Ticket 2 notification, should send 5
##credit.used## = 3
##credit.left##' => $credit['quantity'] -  (ConsumedTicket 1 + ConsumedTicket 2) 
              5           =>              10              -           (2+3) 
                                                              
Looking forward to reading your thoughts.

Reply:
I missed that point.

I would have done it differently (see above) but I do not know which request is easier to understand.
```sql
SELECT
   `glpi_plugin_credit_entities`.`name`,
   `glpi_plugin_credit_entities`.`quantity`,
   (SELECT SUM(`glpi_plugin_credit_tickets`.`consumed`) FROM `glpi_plugin_credit_tickets` WHERE `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = `glpi_plugin_credit_entities`.`id` AND `glpi_plugin_credit_tickets`.`tickets_id` = {$id}) AS `consumed_on_ticket`,
   (SELECT SUM(`glpi_plugin_credit_tickets`.`consumed`) FROM `glpi_plugin_credit_tickets` WHERE `glpi_plugin_credit_tickets`.`plugin_credit_entities_id` = `glpi_plugin_credit_entities`.`id`) AS  `consumed_total`
FROM
   `glpi_plugin_credit_entities`
```

I think your request looks complicated to me due to usage of aliases.

@orthagh What is your opinion ?

Reply:
I tried your sql and it works correctly.
Do you want me to change it?

Reply:
@xacobofg Yes, please.

We talk about that with @trasher and we agree that my proposal will be easier to handle with new querying syntax (even if it is not yet possible).